third_party/move/tools/move-package-manifest/src/manifest.rs

```diff
@@ -0,0 +1,328 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{NamedAddress, PackageName};
+use move_core_types::account_address::AccountAddress;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::{
+    collections::{BTreeMap, BTreeSet},
+    str::FromStr,
+};
+use url::Url;
+
+/***************************************************************************************************
+ * Manifest Definition
+ *
+ **************************************************************************************************/
+/// Represents the full parsed contents of a `Move.toml` manifest file.
+#[derive(Default, Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct PackageManifest {
+    /// Metadata about the package itself, such as package name etc.
+    pub package: PackageMetadata,
+
+    /// Named address mappings defined by the package.
+    #[serde(default)]
+    pub addresses: BTreeMap<NamedAddress, AddressOrWildcard>,
+
+    /// Dev-only named address defined by the package.
+    #[serde(default, rename = "dev-addresses")]
+    pub dev_addresses: BTreeMap<NamedAddress, AccountAddress>,
+
+    /// Build options.
+    pub build: Option<BuildOptions>,
+
+    /// Regular (non-dev) package dependencies.
+    #[serde(default)]
+    pub dependencies: BTreeMap<PackageName, Dependency>,
+
+    /// Dev-only package dependencies.
+    #[serde(default, rename = "dev-dependencies")]
+    pub dev_dependencies: BTreeMap<PackageName, Dependency>,
+}
+
+/// Metadata defined in the `[package]` section of `Move.toml`.
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct PackageMetadata {
+    /// Named of the Move package.
+    pub name: PackageName,
+
+    /// Version of the package.
+    pub version: Version,
+
+    /// List of authors.
+    #[serde(default)]
+    pub authors: BTreeSet<String>,
```

Comment:
Should authors be a set or a list? If set, it is a bit counterintuitive that the author list ["alice", "alice"] doesn't give any kind of warning.